NYPL-Simplified/circulation-patron-web
# Load embedded EPubs in webpub-viewer

This PR updates the NextJS app to load the `webpub-viewer` to render Exploded EPub files.

When a link to a specific `container.xml` file for an Exploded EPub is passed to the viewer as a query param it dynamically displays the rendered e-book. 

The app should redirect to the reader for a specific book when the book type is `"application/vnd.librarysimplified.axisnow+json"`  (or `link.type === "application/vnd.librarysimplified.web-epub"` this is currently commented out as it won't be a valid type in production but is required for the current dev test server for this feature).

To test: 

1: Run the app with `SIMPLIFIED_CATALOG_BASE={ENTER TEST OPDS FEED HERE} SHORTEN_URLS=false npm run dev` using the test OPDS feed.
2. Go to http://localhost:3000/
3. Click "plain text" (unecrypted) e-book which has type `link.type === "application/vnd.librarysimplified.web-epub"` (make sure that check isn't commented out in FulfillmentCard.tsx)
4. Click download link
5. Redirected to `webpub-viewer` with the desired e-book rendered (and updated URL)

Webpub-viewer loaded from c-p-w: ![Screen Shot 2020-07-15 at 3 25 02 PM](https://user-images.githubusercontent.com/6998954/87586912-5c663500-c6af-11ea-9a71-0fa13776f2a5.png)


Note: The `webpub-viewer` currently only supports rendering unencrypted books. There is current working being done for `webpub-viewer` to support decrypted supported e-books.
